event-streams/src/event-streams-v1.0-wd01.md

```diff
@@ -0,0 +1,547 @@
+supported by the source (the event log node). This specification defines two
+simplified filters that allow for a conforming implementation without having to
+implement the full [AMQP filter expressions](#filtex-v10) specification in a
+conforming fashion.
+
+The source filter MUST be evaluated against the retained events in the log from
+earliest to latest. Events matching the source expression are eligible for
+delivery.
+
+# 2.2.2.1 event-streams-delivery-annotations filter
+
+The `event-streams-delivery-annotations` filter type applies to AMQP
+delivery annotations section (AMQP 3.2.2). The filter evaluates to true if all
+annotations enclosed in the filter expression match the respective
+delivery annotation map entries in the message.
+
+Only the [`event-streams-offset`](#2211-event-streams-offset) and/or
+[`event-streams-timestamp`](#2212-event-streams-timestamp) delivery annotation
+properties are eligible to be used with this filter.
+
+The property values match the given values in the filter expression, if the value
+of the property is greater or equal to the value given in the filter expression.
+
+``` XML
+<type name="event-streams-delivery-annotations-filter" class="restricted" source="amqp:map" provides="filter">
+<descriptor name="amqp:event-streams-delivery-annotations-filter" code="0x00000000:0x00000200"/>
+</type> 
+```
+
+The filter-type is “amqp:event-streams-delivery-annotations-filter”, with
+type-code 0x00000000:0x00000200
+
+# 2.2.2.2 event-streams-sql filter 
+
+The event streams SQL filter expression is a constrained subset of the SQL filter
+expression defined in [AMQP filter expressions [FILTEX, 6]](#filtex-v10).
+
+The grammar builds on the grammar defined for those SQL filters, but is
+constrained as follows:
+
+```
+predicate ::=  <boolean_constant> |
+              | <expression> <comparison-operator> <expression>
+              | <predicate> <binary-logical-operator> <predicate>
+expression ::= <constant> | <field_value>  
+constant   ::= <string_constant> | <integer_constant>
+field_value := <field>
+field ::= <section> ’.’<field_name>
+section ::= { ‘message_annotations’ | ‘m’ }
+```
+
+This grammar subset allows for a minimal implementation that is limited to
+evaluating the delivery annotation properties defined here while being
+interoperable with the full SQL filter syntax.
+
+Examples:
+
+- Start from event at offset "a4c5" exclusively: `m.event-streams-offset > '100'`
+- Start from event at offset "a4c5" inclusively: `m.event-streams-offset >= '100'`
+- Start from event received after the timestamp: `m.event-streams-timestamp > 1585672841`
+
+
+``` XML
+ <type name="event-streams-sql-filter" class="restricted" source="sql-filter" provides="filter">
+<descriptor name="amqp:event-streams-sql-filter" code="0x00000000:0x00000201"/> 
+</type>
+```
+
+The filter-type is “amqp:event-streams-sql-filter”, with type-code 0x00000000:0x00000201
```

Comment:
This looks good, but it seems to target consumers which do their own offset tracking. What would be the way to deal with event-log-nodes which track the offsets for the clients? In that case, we might need to be able to indicate where does the consumer want to start if there is no previous position as opposed to always telling the event log node where to start.